File: apps/web/src/api/catalog.ts
import { useQuery } from '@tanstack/react-query'
import { apiClient } from './client'

export interface Brand {
  id: string
  name: string
  slug: string
  country?: string | null
  founded?: number | null
  color?: string | null
  primary_category: string
  categories?: string[] | null
  logo_path?: string | null
  website?: string | null
  description?: string | null
  is_active: boolean
  sort_order: number
}

export interface ClubModel {
  id: string
  brand_id: string
  name: string
  club_type: string
  year?: number | null
  default_loft?: number | null
  default_lie?: number | null
  is_active: boolean
}

export interface ConnectorCatalog {
  id: string
  name: string
  description?: string | null
  status: 'available' | 'coming_soon' | string
  capabilities?: string[] | null
  color?: string | null
  logo_path?: string | null
  website?: string | null
}

export function useBrands(category?: string) {
  return useQuery({
    queryKey: ['catalog', 'brands', category ?? null],
    queryFn: () => {
      const qs = category ? `?category=${encodeURIComponent(category)}` : ''
      return apiClient.get<Brand[]>(`/catalog/brands${qs}`)
    },
    staleTime: 1000 * 60 * 30,
  })
}

export function useClubModels(params?: { brandId?: string; clubType?: string; year?: number }) {
  const qs = new URLSearchParams()
  if (params?.brandId) qs.set('brand_id', params.brandId)
  if (params?.clubType) qs.set('club_type', params.clubType)
  if (params?.year) qs.set('year', String(params.year))
  const query = qs.toString()
  const endpoint = `/catalog/club-models${query ? `?${query}` : ''}`

  return useQuery({
    queryKey: ['catalog', 'club-models', params ?? null],
    queryFn: () => apiClient.get<ClubModel[]>(endpoint),
    enabled: !!params?.brandId,
    staleTime: 1000 * 60 * 30,
  })
}

export function useConnectorsCatalog() {
  return useQuery({
    queryKey: ['catalog', 'connectors'],
    queryFn: () => apiClient.get<ConnectorCatalog[]>('/catalog/connectors'),
    staleTime: 1000 * 60 * 30,
  })
}

File: apps/api/app/schemas/catalog.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, ConfigDict


class BrandResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    slug: str
    country: Optional[str] = None
    founded: Optional[int] = None
    color: Optional[str] = None
    primary_category: str = "clubs"
    categories: Optional[List[str]] = None
    logo_path: Optional[str] = None
    website: Optional[str] = None
    description: Optional[str] = None
    is_active: bool = True
    sort_order: int = 0


class ClubModelResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    brand_id: str
    name: str
    club_type: str
    year: Optional[int] = None
    default_loft: Optional[float] = None
    default_lie: Optional[float] = None
    is_active: bool = True


class ConnectorCatalogResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    description: Optional[str] = None
    status: str = "available"
    capabilities: Optional[List[str]] = None
    color: Optional[str] = None
    logo_path: Optional[str] = None
    website: Optional[str] = None
